diesel/src/migrations/mod.rs

```diff
@@ -91,27 +93,128 @@ use std::path::{PathBuf, Path};
 ///
 /// See the [module level documentation](index.html) for information on how migrations should be
 /// structured, and where Diesel will look for them by default.
-pub fn run_pending_migrations<Conn>(conn: &Conn) -> Result<(), RunMigrationsError> where
+pub fn run_pending_migrations<Conn>(conn: &Conn) -> Result<Vec<String>, RunMigrationsError> where
     Conn: MigrationConnection,
 {
     let migrations_dir = try!(find_migrations_directory());
     run_pending_migrations_in_directory(conn, &migrations_dir, &mut stdout())
 }
 
+/// Generates a module to run migrations on startup. This function will return an `Err` if some 
+/// error occurs reading the migrations. The generated module will be written to `dst`. The 
+/// generated module will contain all migrations that are currently located in the migrations directory. 
+///
+/// # Generate the migration module.
+/// Run this function form `build.rs` to generate the migrations module.
+/// 
+/// ```rust
+/// extern crate diesel;
+/// # include!("src/doctest_setup.rs");
+/// 
+/// use std::path::PathBuf;
+/// use std::env;
+/// use diesel::migrations::generate_migrations_module;
+/// 
+/// fn main() {
+///     let dst = PathBuf::from(env::var_os("OUT_DIR").unwrap());
+///     generate_migrations_module(dst.join("migrations.rs").as_path()).unwrap();
+/// }
+/// ```
+/// # Using the migrations module
+/// To use the generated migration module include it into a file. To run all pending migrations execute
+/// the following code. `migrations::run()` will print all progress to
+/// stdout. This function will return an `Err` if some error occurs reading the migrations, or if
+/// any migration fails to run. Each migration is run in its own transaction, so some migrations
+/// may be committed, even if a later migration fails to run. If there was no Err a Vec with versioncodes 
+/// of executed migrations will be returned.
+/// 
+/// ```rust
+/// # extern crate diesel;
+/// # include!("src/doctest_setup.rs");
+/// #
+/// include!(concat!(env!("OUT_DIR"), "/migrations.rs"));
+/// 
+/// use self::migrations::run;
+/// let connection = establish_connection();
+/// run(&conn).unwrap();
+/// ```
+/// See the [module level documentation](index.html) for information on how migrations should be
+/// structured, and where Diesel will look for them by default.
+
+pub fn generate_migrations_module(dst: &Path) -> Result<(), RunMigrationsError> {
+    let migration_module=try!(build_mirgations_module());
+    let mut f = try!(File::create(dst));
+    try!(f.write_all(&migration_module.as_bytes()));
+    Ok(())
+}
+
+
 #[doc(hidden)]
-pub fn run_pending_migrations_in_directory<Conn>(conn: &Conn, migrations_dir: &Path, output: &mut Write)
-    -> Result<(), RunMigrationsError> where
+pub fn run_pending_migrations_form_iter<Conn, T, M>(conn: &Conn, all_migrations:T, output: &mut Write)-> Result<Vec<String>, RunMigrationsError> where
```

Comment:
Please make sure the code adheres to the [style guidelines](https://github.com/sgrif/diesel/issues/120#issuecomment-174222639)